spark3-extensions/src/main/antlr/org.apache.spark.sql.catalyst.parser.extensions/IcebergSqlExtensions.g4

```diff
@@ -69,6 +69,7 @@ statement
     : CALL multipartIdentifier '(' (callArgument (',' callArgument)*)? ')'                  #call
     | ALTER TABLE multipartIdentifier ADD PARTITION FIELD transform (AS name=identifier)?   #addPartitionField
     | ALTER TABLE multipartIdentifier DROP PARTITION FIELD transform                        #dropPartitionField
+    | ALTER TABLE multipartIdentifier REPLACE PARTITION FIELD transform TO transform (AS name=identifier)? #replacePartitionField
```

Comment:
`REPLACE ... TO` doesn't make sense. What about `REPLACE ... WITH` instead?

Reply:
Also, why support `AS`? Shouldn't the new partition field use the same name as the old partition field?

I would also expect using a partition field name to work. So if I used `bucket(16, id) AS shard` to create the partition, then I should also be able to use `REPLACE shard WITH bucket(32, id)`.

Reply:
Looks like the implementation supports looking up the field by name, and `AS` can support rename, like `REPLACE ts_day WITH hour(ts) AS ts_hour`.

Reply:
I think there are 2 use cases that have contradicting behaviors:
1. `ADD PARTITION FIELD bucket(id, 16) AS shard`, then `REPLACE PARTITION FIELD shard WITH bucket(id, 32)`
2. `ADD PARTITION FIELD days(ts) AS days_col`, then `REPLACE PARTITION FIELD days_col WITH hours(ts)`

For case 1, we do want the `bucket(id, 32)` to also be called `shard`, but we don't really want to call the `hours(ts)` partition as `days_col`. 

So here are a couple of observations for `REPLACE transformFrom WITH transformTo`:
1. if `transformFrom` is an expression, the default partition field has very specific meanings such as `ts_days`, `id_bucket_16`, and the replaced partition field `transformTo` should not inherit that name
2. if there is a custom name for the `transformFrom` partition field, the behavior really depends. The 2 examples above shows this contradicting expectations.

So I think the safest approach is to not infer the behavior for the custom partition name. If the caller wants to use the same name, just use the `AS` clause to specify it again, such as `REPLACE PARTITION FIELD shard WITH bucket(id, 32) AS shard`.

What do you think?

Reply:
Currently we strictly disallow rename -> delete renamed field -> add field with the same name in a single commit for use case 1, but I think that sounds like something we can support in `BaseUpdatePartitionSpec`, I am thinking about any consequence of it, will update later.

Reply:
I think you make a good argument that names should not be reused. If the caller wants to use the same name, then it isn't hard to put the name twice: `REPLACE shard WITH bucket(32, id) AS shard`. Then all we need to support is drop and add with the same name. I think that's simpler to implement and works well.